Source/JavaScriptCore/wasm/js/WebAssemblyFunction.cpp

```diff
@@ -86,20 +86,20 @@ JSC_DEFINE_HOST_FUNCTION(callWebAssemblyFunction, (JSGlobalObject* globalObject,
         boxedArgs.append(JSValue::decode(value));
     }
 
-    // When we don't use fast TLS to store the context, the JS
-    // entry wrapper expects a JSWebAssemblyInstance as the |this| value argument.
-    JSValue context = instance;
     JSValue* args = boxedArgs.data();
     int argCount = boxedArgs.size() + 1;
 
     // Note: we specifically use the WebAssemblyFunction as the callee to begin with in the ProtoCallFrame.
     // The reason for this is that calling into the llint may stack overflow, and the stack overflow
     // handler might read the global object from the callee.
+    // For wasm, we setup |codeBlock| and |this| differently.
+    //     |this| : We keep JSWebAssemblyInstance in |this| to anchor it for GC.
+    //     |codeBlock| : JS entry wrapper expects a Wasm::Instance* as the |codeBlock| value argument.
     ProtoCallFrame protoCallFrame;
-    protoCallFrame.init(nullptr, globalObject, wasmFunction, context, argCount, args);
+    protoCallFrame.init(nullptr, globalObject, wasmFunction, instance, argCount, args);
```

Comment:
Can we also rename `instance` to `jsInstance` (maybe in a follow up patch)?  It's pretty confusing to use the same `instance` name to mean different things in different places.